### PR TITLE
 Add ability of marshalling messages sent back to `sender`

### PR DESCRIPTION
**Pull request by sergiy0**

### CODE REVIEW

File: safe-actor/src/main/scala/com/evolutiongaming/safeakka/actor/Sender.scala [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+object Sender {
+
+  private case class SenderImpl(protected val ref: ActorRef) extends Sender {
+    
+    override def ![A](msg: A)(implicit sender: ActorRef = Actor.noSender, marshalReply: Sender.MarshalReply[A]): Unit =
+      ref.!(marshalReply marshal msg)(sender)
+
+    def path: ActorPath = ref.path
+
+    def ==(that: ActorRef): Boolean = ref equals that
```
Comment (t3hnar) [inline safe-actor/src/main/scala/com/evolutiongaming/safeakka/actor/Sender.scala]: do we really need this ?

File: safe-actor/src/main/scala/com/evolutiongaming/safeakka/actor/Sender.scala [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+  def apply(actorRef: ActorRef): Sender = SenderImpl(actorRef)
+
+  val Empty: Sender = Sender(ActorRef.noSender)
+
+  trait MarshalReply[-A] {
+    def marshal: A => Any
+  }
+
+  lazy val TestIdentityMarshaller: MarshalReply[Any] = new MarshalReply[Any] { def marshal = identity }
```
Comment (t3hnar) [inline safe-actor/src/main/scala/com/evolutiongaming/safeakka/actor/Sender.scala]: why `Test` as prefix ? 
I'd name it as `MarshalReply.AnyImpl` or `MarshalReply.Any`

**Comment by sergiy0** [comment]
@t3hnar Fixed.

File: safe-actor/src/main/scala/com/evolutiongaming/safeakka/actor/Sender.scala [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    def path: ActorPath = ref.path
+  }
+
+  def apply(actorRef: ActorRef): Sender = SenderImpl(actorRef)
+
+  val Empty: Sender = Sender(ActorRef.noSender)
+
+  trait MarshalReply[-A] {
+    def marshal: A => Any
```
Comment (t3hnar) [inline safe-actor/src/main/scala/com/evolutiongaming/safeakka/actor/Sender.scala]: `marshal`  -> `apply` ?

Comment (t3hnar) [reply safe-actor/src/main/scala/com/evolutiongaming/safeakka/actor/Sender.scala]: but why do you need  `marshalReply` at all ? why not just add add `map` method to `Sender` ?

**Pull request closed without merging**